gateway-ha/src/main/java/io/trino/gateway/ha/handler/ProxyUtils.java

```diff
@@ -0,0 +1,164 @@
+     * This regular expression matches query ids as they appear in the path of a URL. The query id must be preceded
+     * by a "/". A query id is defined as three groups of digits separated by underscores, with a final group
+     * consisting of any alphanumeric characters.
+     */
+    private static final Pattern QUERY_ID_PATH_PATTERN = Pattern.compile(".*/(\\d+_\\d+_\\d+_\\w+).*");
+    /**
+     * This regular expression matches query ids as they appear in the query parameters of a URL. The query id is
+     * defined as in QUERY_TEXT_LENGTH_FOR_HISTORY. The query id must either be at the beginning of the query parameter
+     * string, or be preceded by %2F (a URL-encoded "/"), or  "query_id=", with or without the underscore and any
+     * capitalization.
+     */
+    private static final Pattern QUERY_ID_PARAM_PATTERN = Pattern.compile(".*(?:%2F|(?i)query_?id(?-i)=|^)(\\d+_\\d+_\\d+_\\w+).*");
+    private static final Pattern EXTRACT_BETWEEN_SINGLE_QUOTES = Pattern.compile("'([^\\s']+)'");
+
+    private ProxyUtils() {}
+
+    public static String getQueryUser(String userHeader, String authorization)
+    {
+        if (!isNullOrEmpty(userHeader)) {
+            log.debug("user from %s", USER_HEADER);
+            return userHeader;
+        }
+
+        log.debug("user from basic auth");
+        String user = "";
+        if (authorization == null) {
+            log.debug("didn't find any basic auth header");
+            return user;
+        }
+
+        int space = authorization.indexOf(' ');
+        if ((space < 0) || !authorization.substring(0, space).equalsIgnoreCase("basic")) {
+            log.error("basic auth format is incorrect");
+            return user;
+        }
+
+        String headerInfo = authorization.substring(space + 1).trim();
+        if (isNullOrEmpty(headerInfo)) {
+            log.error("The encoded value of basic auth doesn't exist");
+            return user;
+        }
+
+        String info = new String(Base64.getDecoder().decode(headerInfo));
+        List<String> parts = Splitter.on(':').limit(2).splitToList(info);
+        if (parts.size() < 1) {
+            log.error("no user inside the basic auth text");
+            return user;
+        }
+        return parts.get(0);
+    }
+
+    public static String extractQueryIdIfPresent(HttpServletRequest request)
+    {
+        String path = request.getRequestURI();
+        String queryParams = request.getQueryString();
+        try {
+            String queryText = CharStreams.toString(new InputStreamReader(request.getInputStream()));
+            if (!isNullOrEmpty(queryText)
+                    && queryText.toLowerCase().contains("system.runtime.kill_query")) {
+                // extract and return the queryId
+                String[] parts = queryText.split(",");
+                for (String part : parts) {
+                    if (part.contains("query_id")) {
+                        Matcher m = EXTRACT_BETWEEN_SINGLE_QUOTES.matcher(part);
```

Comment:
I think its better to use longer meaningful variable names even in this case. 

`Matcher matcher`